test/yagna/helpers/negotiation.py

```diff
@@ -0,0 +1,112 @@
+"""Helper functions for building custom Offers and negotiating Agreements."""
+
+import logging
+from typing import List, Optional, Callable, Tuple, Any
+from datetime import datetime, timedelta
+
+from ya_market import Demand, DemandOfferBase, Proposal
+
+from goth.runner.probe import RequestorProbe
+from goth.runner.provider import ProviderProbeWithLogSteps
+from goth.runner.requestor import RequestorProbeWithApiSteps
+
+
+logger = logging.getLogger(__name__)
+
+
+class DemandBuilder:
+    """Helper for building custom Demands.
+
+    Use if RequestorProbeWithApiSteps.subscribe_template_demand function
+    is not enough for you.
+    """
+
+    def __init__(self, requestor: RequestorProbe):
+        self._requestor = requestor
+        self._properties = dict()
+        self._constraints = "()"
+
+    def props_from_template(self, task_package: str) -> "DemandBuilder":
+        """Build default properties."""
+
+        new_props = {
+            "golem.node.id.name": "test-requestor-{}".format(self._requestor.name),
```

Comment:
*f-strings* are used elsewhere in `goth` codebase:
```suggestion
            "golem.node.id.name": f"test-requestor-{self._requestor_name}",
```